pkg/controller/certificates/signer/cfssl_signer.go

```diff
@@ -1,153 +0,0 @@
-/*
-Copyright 2016 The Kubernetes Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-// Package signer implements a CA signer that uses keys stored on local disk.
-package signer
-
-import (
-	"crypto"
-	"crypto/x509"
-	"fmt"
-	"io/ioutil"
-	"os"
-	"time"
-
-	capi "k8s.io/api/certificates/v1beta1"
-	certificatesinformers "k8s.io/client-go/informers/certificates/v1beta1"
-	clientset "k8s.io/client-go/kubernetes"
-	"k8s.io/kubernetes/pkg/controller/certificates"
-
-	"github.com/cloudflare/cfssl/config"
-	"github.com/cloudflare/cfssl/helpers"
-	"github.com/cloudflare/cfssl/signer"
-	"github.com/cloudflare/cfssl/signer/local"
-)
-
-func NewCSRSigningController(
-	client clientset.Interface,
-	csrInformer certificatesinformers.CertificateSigningRequestInformer,
-	caFile, caKeyFile string,
-	certificateDuration time.Duration,
-) (*certificates.CertificateController, error) {
-	signer, err := newCFSSLSigner(caFile, caKeyFile, client, certificateDuration)
-	if err != nil {
-		return nil, err
-	}
-	return certificates.NewCertificateController(
-		"csrsigning",
-		client,
-		csrInformer,
-		signer.handle,
-	), nil
-}
-
-type cfsslSigner struct {
-	ca                  *x509.Certificate
-	priv                crypto.Signer
-	sigAlgo             x509.SignatureAlgorithm
-	client              clientset.Interface
-	certificateDuration time.Duration
-
-	// nowFn returns the current time.  We have here for unit testing
-	nowFn func() time.Time
-}
-
-func newCFSSLSigner(caFile, caKeyFile string, client clientset.Interface, certificateDuration time.Duration) (*cfsslSigner, error) {
-	ca, err := ioutil.ReadFile(caFile)
-	if err != nil {
-		return nil, fmt.Errorf("error reading CA cert file %q: %v", caFile, err)
-	}
-	cakey, err := ioutil.ReadFile(caKeyFile)
-	if err != nil {
-		return nil, fmt.Errorf("error reading CA key file %q: %v", caKeyFile, err)
-	}
-
-	parsedCa, err := helpers.ParseCertificatePEM(ca)
-	if err != nil {
-		return nil, fmt.Errorf("error parsing CA cert file %q: %v", caFile, err)
-	}
-
-	strPassword := os.Getenv("CFSSL_CA_PK_PASSWORD")
```

Comment:
This is an external-facing change. Do we need to preserve this?

Reply:
I would be ok with a release note if you think anyone is actually using this. This is undocumented behavior, and it seems like it would be unlikely. Happy to preserve it.

Reply:
added in https://github.com/kubernetes/kubernetes/commit/fb099ae3856750ddad24a8cc5742b515d033c75d#diff-f0a1084e82f4fa0dd67b7d41dd1d3fd6R63, doesn't look like it was prompted by a specific use case.

also called out as problematic in https://github.com/kubernetes/kubernetes/issues/81117

I'm fine with a release note and removal

Reply:
Done.